### PR TITLE
VDBOX+SFC for 1->1 transcode scenarios

### PR DESCRIPTION
**Pull request by dmitryermilov**
The changes introduces supports of 1->1, 1->N transcode with SFC (previously
only N->1 was supported).
Plus it fixes the bug on library side where decoders couldn't accept surfaces
allocated in addition to what decoder::QueryIOSurf() requested.

Test: ./sample_multi_transcode -dec_postproc -i::h264 in.h264 -o::h264 out.h264 -w 1280 -h 720 -hw

or via par file
-dec_postproc -i::h264 in.h264 -o::h264 out.h264 -w 1280 -h 720 -hw -o::sink
-i::source -o::h265 o1.h265
-i::source -o::h265 o2.h265

Signed-off-by: Dmitry Ermilov <dmitry.ermilov@intel.com>

### CODE REVIEW

File: _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp [hunk 1]
```diff
@@ -93,9 +93,9 @@ struct ThreadTaskInfo
     mfxFrameSurface1* surface_work = nullptr;
     mfxFrameSurface1* surface_out  = nullptr;
 
-    ThreadTaskInfo(mfxFrameSurface1* work, mfxFrameSurface1* out)
-        : surface_work(work)
-        , surface_out(out)
+    ThreadTaskInfo(mfxFrameSurface1* work, mfxFrameSurface1* out)
+        : surface_work(work)
+        , surface_out(out)
```
Comment (wangyan-intel) [inline _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp:98]: Sorry. I cannot find what are changed? And it seems not be related with "VDBOX+SFC for 1->1 transcode scenarios". Could we add it into another commit?

File: _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp [hunk 2]
```diff
@@ -1597,11 +1592,7 @@ mfxStatus VideoDECODEH264::DecodeFrame(mfxFrameSurface1 *surface_out, UMC::H264D
     {
         index = m_FrameAllocator->FindSurface(surface_out, m_isOpaq);
         pFrame = m_pH264VideoDecoder->FindSurface((UMC::FrameMemID)index);
-        if (!pFrame)
-        {
-            VM_ASSERT(false);
-            return MFX_ERR_NOT_FOUND;
-        }
```
Comment (wangyan-intel) [inline _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp:1604]: it seems not be related with "VDBOX+SFC for 1->1 transcode scenarios". Could we add it into another commit?

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 3]
```diff
@@ -1399,11 +1399,7 @@ mfxStatus VideoDECODEH265::DecodeFrame(mfxFrameSurface1 *surface_out, H265Decode
     {
         index = m_FrameAllocator->FindSurface(surface_out, m_isOpaq);
         pFrame = m_pH265VideoDecoder->FindSurface((UMC::FrameMemID)index);
-        if (!pFrame)
-        {
-            VM_ASSERT(false);
-            MFX_RETURN(MFX_ERR_NOT_FOUND);
-        }
```
Comment (wangyan-intel) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp:1406]: Same the above comments.

**Review by wangyan-intel — CHANGES_REQUESTED** [review]

**Review by AntonGrishin — DISMISSED** [review]
Ok, from the sample side

File: _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp (hunk 2 above)
Comment (dmitryermilov) [reply _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp:1604]: addressed

**Review by wangyan-intel — APPROVED** [review]
LGTM

**Pull request merged**